index.js

```diff
@@ -0,0 +1,135 @@
+const env = process.env;
+
+function matchExact(r, str) {
+  let match = str.match(r);
+  return match && str === match[0];
+}
+
+function fixNewLineEOF(b) {
+  // If file is empty
+  if (b.length < 1) {
+    return b;
+  }
+  // If file is not empty
+  b = b.replace(/[ \t\n]*$/, '\n');
+
+  if (b.length === 1) {
+    // if the remaining character is a whitespace
+    if (/[ \t\n]*$/.test(b)) {
+      return '';
+    } else {
+      return b + '\n';
+    }
+  }
+  return b;
+}
+
+async function getChangedFilesPaths(pull_request, octokit, owner, repo) {
+  const { data: pullRequestDiff } = await octokit.pulls.get({
+    owner,
+    repo,
+    pull_number: pull_request.number,
+    mediaType: {
+      format: 'diff'
+    }
+  });
+
+  const parsedDiff = gitDiffParser(pullRequestDiff);
+
+  const changedFilePaths = parsedDiff.map((e) => {
+    return e['newPath'].replace('b/', '');
+  });
+
+  return changedFilePaths;
+}
+
+async function run() {
+  const token = core.getInput('GH_TOKEN');
+  let ignorePaths = core.getInput('IGNORE_PATHS');
+  if (!ignorePaths) {
+    ignorePaths = [];
+  } else {
+    ignorePaths = ignorePaths.split(' ');
+  }
+
+  core.info('ignorePaths' + JSON.stringify(ignorePaths));
+
+  let branch;
+  if (github.context.eventName == 'pull_request') {
+    branch = github.context.payload.pull_request.head.ref;
+  } else {
+    branch = github.context.ref.replace('refs/heads/', '');
+  }
+
+  const git = simpleGit();
+
+  const octokit = github.getOctokit(token);
+  const { context = {} } = github;
+  const { pull_request } = context.payload;
+
+  const owner = env.GITHUB_REPOSITORY.split('/')[0];
+  const repo = env.GITHUB_REPOSITORY.split('/')[1];
+
+  const changedFilePaths = await getChangedFilesPaths(
+    pull_request,
+    octokit,
+    owner,
+    repo
+  );
+
+  core.info('changedFilePaths ' + JSON.stringify(changedFilePaths));
+
+  // Removec files matching ignore paths regex
+  let filesToCheck = changedFilePaths.map((e) => {
+    for (let i = 0; i < ignorePaths.length; i++) {
+      if (matchExact(ignorePaths[i], e)) {
+        return null;
+      }
+    }
+    return e;
+  });
+
+  core.info('filesToCheck ' + JSON.stringify(filesToCheck));
+
+  // Perform EOF newline check
+  for (let i = 0; i < filesToCheck.length; i++) {
+    if (filesToCheck[i] !== null) {
+      let data = fs.readFileSync(filesToCheck[i], {
+        encoding: 'utf8',
+        flag: 'r'
+      });
+      data = fixNewLineEOF(data);
+      fs.writeFileSync(filesToCheck[i], data, 'utf8');
+    }
+  }
+
+  // Generate DIff and commit changes
+  const diff = await exec.exec('git', ['diff', '--quiet'], {
+    ignoreReturnCode: true
+  });
+
+  core.info(JSON.stringify(diff));
+
+  if (diff) {
+    await core.group('push changes', async () => {
+      await git.addConfig(
+        'user.email',
+        `${env.GITHUB_ACTOR}@users.noreply.github.com`
+      );
+      await git.addConfig('user.name', env.GITHUB_ACTOR);
+      await git.add('./');
```

Comment:
Won't this blindly add all files? Can we instead add only files we know we changed?